web/src/activity.ts

```diff
@@ -191,16 +193,54 @@ export function mark_client_active(): void {
     }
 }
 
+export function setup_browser_idle_detector(idle_handler: {cancel: () => void}): void {
+    // This code is separated out of `initialize` to facilitate testing
+    if (!browser_idle_detection.is_browser_idle_detector_supported()) {
+        blueslip.log("Browser idle detector not supported");
+        return;
+    }
+
+    $(document).one("keypress click", () => {
+        void browser_idle_detection.request_browser_idle_detector_permission();
```

Comment:
Similarly, `.request_persmission()`` would mimic other APIs like the Notification API

`let promise = Notification.requestPermission();`

We're kinda limited to snake_case for our own functions, but still let's try to be as concise as reasonable.